### PR TITLE
SB-29553 - Added mapping reqired for user and org enhancements

### PR DESCRIPTION
**Pull request by karthik-tarento**

### CODE REVIEW

**Comment by karthik-tarento** [comment]
@reshmi-nair -- created PR as requested for SB-29553 story. This PR contains ES mappings for user and org.
Please review and merge

File: ansible/roles/es-mapping/files/mappings/userv3-mapping.json [hunk 1]
```diff
@@ -889,6 +889,937 @@
           }
         }
       }
+    },
+    "profileDetails": {
```
Comment (amit-tarento) [inline ansible/roles/es-mapping/files/mappings/userv3-mapping.json:893]: Please take a look for the below and change accordingly
 "userType": {
      "type": "text",
      "fields": {
        "raw": {
          "type": "text",
          "analyzer": "keylower"
        }
      },
      "analyzer": "cs_search_analyzer",
      "search_analyzer": "cs_search_analyzer"
    },
    
    
    remove copy_to and fielddata= true as its not required

File: ansible/roles/es-mapping/files/mappings/userv3-mapping.json [hunk 2]
```diff
@@ -889,6 +889,937 @@
...
+      "properties": {
+        "photo": {
```
Comment (amit-tarento) [inline ansible/roles/es-mapping/files/mappings/userv3-mapping.json:895]: Do we need this field to be index, Is this field searchable and whats the significance for searching this field.

File: ansible/roles/es-mapping/files/mappings/userv3-mapping.json [hunk 3]
```diff
@@ -889,6 +889,937 @@
...
+            "all_fields"
+          ],
+          "analyzer": "cs_index_analyzer",
+          "search_analyzer": "cs_search_analyzer",
+          "fielddata": true
+        },
+        "personalDetails": {
+          "properties": {
+            "primaryEmail": {
+              "type": "text",
```
Comment (amit-tarento) [inline ansible/roles/es-mapping/files/mappings/userv3-mapping.json:914]: Do we need this email field as text , Do we really need this field to be analyzed or apply fuzzy or auto search capability

File: ansible/roles/es-mapping/files/mappings/userv3-mapping.json [hunk 4]
```diff
@@ -889,6 +889,937 @@
...
+              },
+              "copy_to": [
+                "all_fields"
+              ],
+              "analyzer": "cs_index_analyzer",
+              "search_analyzer": "cs_search_analyzer",
+              "fielddata": true
+            },
+            "maritalStatus": {
+              "type": "text",
```
Comment (amit-tarento) [inline ansible/roles/es-mapping/files/mappings/userv3-mapping.json:994]: same as above comment for email field.

File: ansible/roles/es-mapping/files/mappings/userv3-mapping.json [hunk 5]
```diff
@@ -889,6 +889,937 @@
...
+              "copy_to": [
+                "all_fields"
+              ],
+              "analyzer": "cs_index_analyzer",
+              "search_analyzer": "cs_search_analyzer",
+              "fielddata": true
+            },
+            "nationality": {
+              "type": "text",
+              "fields": {
```
Comment (amit-tarento) [inline ansible/roles/es-mapping/files/mappings/userv3-mapping.json:1027]: Please take a look again for all these fields and change the data type for fields if its not required to analyzed

**Review by amit-tarento — CHANGES_REQUESTED** [review]

**Comment by karthik-tarento** [comment]
@amit-tarento @reshmi-nair -- closing this PR as we will have forked repo for devOps in our env. Will maintain the user mapping too in our forked repo for now.

**Pull request closed without merging**